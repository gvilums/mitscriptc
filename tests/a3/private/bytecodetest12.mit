f = fun(x) {
  print("world");
  return 1/x;
};

print("hello");
print(f(0));
