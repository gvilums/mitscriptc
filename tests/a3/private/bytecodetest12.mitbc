function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = ["world", 1],
			parameter_count = 1,
			local_vars = [x],
			local_ref_vars = [],
			free_vars = [],
			names = [print],
			instructions = 
			[
				load_global	0
				load_const	0
				call	1
				pop
				load_const	1
				load_local	0
				div
				return
			]
		}
	],
	constants = ["hello", 0],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [f, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	1
		load_func	1
		alloc_closure	0
		store_global	2
		load_func	2
		alloc_closure	0
		store_global	3
		load_func	3
		alloc_closure	0
		store_global	0
		load_global	1
		load_const	0
		call	1
		pop
		load_global	1
		load_global	0
		load_const	1
		call	1
		call	1
		pop
	]
}