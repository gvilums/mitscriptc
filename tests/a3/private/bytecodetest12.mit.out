hello
world
IllegalArithmeticException: Division by zero
